### PR TITLE
[IMP] Auto create partners in event registrations

### PR DESCRIPTION
**Pull request by JavierIniesta**
Auto create partners in event registrations using email as ID.

### CODE REVIEW

**Comment by JavierIniesta** [comment]
cc @rafaelbn

**Comment by janetrp** [comment]
The partner has been created in the event form but not the contact, should the registration email appear in the partner?

2015-11-03 18_26_33-odoo

2015-11-03 18_28_12-odoo

**Comment by JavierIniesta** [comment]
@janetrp , you created the partner manually. If you check the "Create Partners in registration" option, it will make it automatically. Then, it will fill name, email and phone fields.

**Comment by janetrp** [comment]
ok @JavierIniesta got it! Tested and works fine :+1:

**Comment by pedrobaeza** [comment]
Better document that option in the Configuration section of the README

**Comment by JavierIniesta** [comment]
Configuration section added to README

**Comment by rafaelbn** [comment]
Rebuilding for testing! cc @yajo @antespi @sergio-incaser @carlos-incaser @janetrp

File: partner_event/i18n/es.po [hunk 1]
```diff
@@ -16,14 +16,45 @@ msgstr ""
...
+#: model:ir.model,name:partner_event.model_event_registration
+msgid "Event Registration"
+msgstr "Registro a evento"
+
+#. module: partner_event
 #: field:res.partner,attended_registration_count:0
-msgid "Attended registration number"
-msgstr "Nº eventos registrado"
+msgid "Event attended registrations number"
+msgstr "Nº registros a eventos asistidos"
```
Comment (yajo) [inline partner_event/i18n/es.po]: Nº de registrados que han asistido al evento

File: partner_event/i18n/es.po [hunk 2]
```diff
@@ -36,13 +67,21 @@ msgid "Partner"
 msgstr "Empresa"
 
 #. module: partner_event
-#: field:res.partner,registration_count:0
-msgid "Registration number"
-msgstr "Nº registros"
+#: model:ir.actions.act_window,name:partner_event.action_partner_register_event
+msgid "Register in an event"
+msgstr "Registro en un evento"
```
Comment (yajo) [inline partner_event/i18n/es.po]: Registrar

File: partner_event/models/event_registration.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    def _prepare_partner(self, vals):
+        return {
+            'name': vals.get('name') or vals.get('email'),
+            'email': vals.get('email', False),
+            'phone': vals.get('phone', False),
+        }
+
+    @api.model
+    def create(self, vals):
+        if not vals.get('partner_id') and vals.get('email'):
```
Comment (yajo) [inline partner_event/models/event_registration.py:21]: What if there is a name but not an email? Shouldn't it check existence / create the partner?

File: partner_event/models/event_registration.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            partner_model = self.env['res.partner']
+            event_model = self.env['event.event']
+            partner_id = False
+            # Look for a partner with that email
+            partners = partner_model.search(
+                [('email', '=ilike', vals['email'])])
```
Comment (yajo) [inline partner_event/models/event_registration.py]: If you use ilike and do not check the email, this happens:
screenshot from 2015-11-10 10 16 40
screenshot from 2015-11-10 10 16 51

**Comment by yajo** [comment]
Please add tests.

File: partner_event/README.rst [hunk 5]
```diff
@@ -17,6 +17,12 @@ It also includes:
 * Action in partner tree view 'More' button, to register several partners
   to an event
 
+Configuration
+=============
+
+There is a new option in event form view, "Create Partners in registration". If
+this option is checked, when you add registrations to this event, partners will
+be created automatically with name, email and phone fields.
```
Comment (rafaelbn) [inline partner_event/README.rst:26]: Please explain what happens when partner already exists and user only fills email

File: partner_event/models/event_registration.py (hunk 3 above)
Comment (JavierIniesta) [inline partner_event/models/event_registration.py:21]: We can have several partners with the same name. Email is the identity field.

**Comment by JavierIniesta** [comment]
Tests added

**Comment by rafaelbn** [comment]
Tested in runbot :+1:

**Comment by pedrobaeza** [comment]
:+1:

**Pull request merged**